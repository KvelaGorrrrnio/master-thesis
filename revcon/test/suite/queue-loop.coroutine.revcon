@coroutine

coroutine main(int sum, list int result) {
  # Setup
  local queue int queue
  local int x

  # Pre-work
  x += 5
  enqueue x queue
  x += 3
  enqueue x queue
  x += 2
  enqueue x queue
  x += 9
  enqueue x queue
  x += 1
  enqueue x queue
  x += 2
  enqueue x queue
  x += 4
  enqueue x queue

  # Work
  from size queue == 7
  loop
    dequeue x queue
    sum += x
    push x result
  until empty queue

  # Teardown
  delocal int x
  delocal queue int queue
}
