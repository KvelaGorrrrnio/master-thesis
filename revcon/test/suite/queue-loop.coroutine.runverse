test/suite/queue-loop.coroutine.revcon:
[store]
  main: Ã˜
  result: []
  sum: 0

[stack]
[spawn main main end () | unresume main (sum, result) | despawn main main begin ()] <|> []

