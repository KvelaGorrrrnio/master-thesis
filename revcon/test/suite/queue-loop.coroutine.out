test/suite/queue-loop.coroutine.revcon:
[store]
  main: Ã˜
  result: [4, 2, 1, 9, 2, 3, 5]
  sum: 26

[stack]
[spawn main main begin () | resume main (sum, result) | despawn main main end ()] <|> []

